File: client/app/schema/article.ts
import { Category } from "./category";

export interface Article{
    id: number;
    name: string;
    slug: string;
    is_active: boolean;
    image: string | null;
    intro: string | null;
    description: string | null;
    tag: string | null;
    keywords:string | null;
    category: Category;
}

File: server/app/schema/article.py
from pydantic import BaseModel
from typing import Optional
from .category import CategoryResponse

class ArticleBase(BaseModel):
    name: str
    slug: str
    is_active: bool
    intro: Optional[str]
    description: Optional[str]
    tag: Optional[str]
    keywords: Optional[str]
    id_user: int


class ArticleResponse(ArticleBase):
    id: int
    category: Optional[CategoryResponse]
    class Config:
        from_attributes = True

class ArticleCreate(ArticleBase):
    id_category: int
    pass

class ArticleUpdate(ArticleBase):
    id_category: int
    pass
